sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -95,70 +94,68 @@ namespace Azure { namespace Core { namespace Http {
    */
   class HttpMethod final {
   public:
-    HttpMethod() = delete;
```

Comment:
Why did we get rid of this?

Reply:
If custom constructor is present, default constructor does not get generated either way.

Reply:
But I think later we should explicitly list and document all the default operations for all the code. We have a work item for that. We currently don't do that, and it was not worth it to document the deletion of the constructor that happens by the C++ rules anyway.

Reply:
I am planning to not take this to the follow up PR.

Reply:
@CaseyCarter for the extensible enum pattern, should we explicitly delete the default ctor or leave it to the compiler to generate?
The guideline doesn't have it:
https://azure.github.io/azure-sdk/cpp_implementation.html#enumerations

Reply:
I'm perfectly happy for the default constructor to be implicitly deleted without calling it out explicitly. This isn't as likely to cause confusion as are the arcane rules around when move special members are generated or suppressed which motivates the rule of 0/3/5 guideline.